packages/xcoobee-cookie-kit-core/src/LocaleManager.js

```diff
@@ -34,5 +34,8 @@ export function fetchCountryCode() {
       // console.log("LocaleManager#fetchCountryCode fetched.");
       const countryCode = res ? res.countryCode : "US";
       return countryCode;
+    })
+    .catch((error) => {
+      console.error(error);
```

Comment:
This `catch` will make the `catch` in `getCountryCode` sort of unnecessary.

So, to clarify what I was suggesting in the previous review:

This is a lower-level function (at least compared to `getCountryCode`). It is more like a library function.

With library functions, I tend to do one of two things. 1) Make it impossible to throw/reject an error by catching it, then returning some value instead of propagating the error (`null` or `"US"` would be the most appropriate value) or 2) Let the caller of the function call handle any errors.

Which choice would I make? It depends on what the error is and whether the error is something that should be propagated to a higher level. If the error should be made aware to the user, then I would use #2. Or if a reasonable fallback value can't be determined, then use #2.

On some occasions, I might catch specific errors and return some fallback value while letting some errors propagate. Sort of a hybrid of #1 and #2.

With that in mind, I would do one of the following:

A) Return `null` or `"US"` (or pass in the value to be returned when unable to successfully fetch it by adding a parameter to the function) here and remove the unnecessary `catch` on line 212 in the `getCountryCode` method

```js
export function fetchCountryCode(onErrorCountryCode = "US") {
  // console.log("LocaleManager#fetchCountryCode fetching...");
  return fetch("http://ip-api.com/json")
    .then(res => res.json())
    .then((res) => {
      // console.log("LocaleManager#fetchCountryCode fetched.");
      const countryCode = res ? res.countryCode : onErrorCountryCode;
      return countryCode;
    })
    .catch((error) => {
      console.error(error); // Whether we want to display this error to the console is debatable.
      return onErrorCountryCode;
    });
```

B) Remove the `catch` here and make sure all calls to `fetchCountryCode` have their own `catch`.